src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java

```diff
@@ -774,6 +771,22 @@ private PaymentTransactionInfoPlugin executeFollowUpTransaction(final Transactio
         }
     }
 
+    private void updateResponseWithAdditionalProperties(final UUID kbTransactionId, final Iterable<PluginProperty> properties, final UUID tenantId) throws PaymentPluginApiException {
+        try {
+            dao.updateResponse(kbTransactionId, properties, tenantId);
+        } catch (final SQLException e) {
+            throw new PaymentPluginApiException("SQL exception when updating response", e);
+        }
+    }
+
+    private AdyenResponsesRecord fetchResponseIfExist(final UUID kbPaymentId, final UUID tenantId) throws PaymentPluginApiException {
+        try {
+            return dao.getResponses(kbPaymentId, tenantId).stream().findFirst().orElse(null);
```

Comment:
Could you use `getSuccessfulAuthorizationResponse` instead (small perf gain as `limit 1` is done at the DB level)?